oneview/resource_server_certificate.go

```diff
@@ -0,0 +1,533 @@
+			Description:        certificateDetails["description"].(string),
+			Dnqualifier:        certificateDetails["dn_qualifier"].(string),
+			Etag:               certificateDetails["etag"].(string),
+			Email:              certificateDetails["email"].(string),
+			EnhancedKeyUsage:   certificateDetails["enhanced_key_usage"].(string),
+			ExpiresInDays:      certificateDetails["expires_in_days"].(string),
+			GivenName:          certificateDetails["given_name"].(string),
+			Initials:           certificateDetails["initials"].(string),
+			Issuer:             certificateDetails["issuer"].(string),
+			KeyUsage:           certificateDetails["key_usage"].(string),
+			Locality:           certificateDetails["locality"].(string),
+			LocationState:      certificateDetails["location_state"].(string),
+			Modified:           certificateDetails["modified"].(string),
+			Name:               certificateDetails["name"].(string),
+			Organization:       certificateDetails["organization"].(string),
+			OrganizationalUnit: certificateDetails["organizational_unit"].(string),
+			PublicKey:          utils.NewNstring(certificateDetails["public_key"].(string)),
+			SerialNumber:       utils.NewNstring(certificateDetails["serial_number"].(string)),
+			Sha1Fingerprint:    utils.NewNstring(certificateDetails["sha1_fingerprint"].(string)),
+			Sha256Fingerprint:  utils.NewNstring(certificateDetails["sha256_fingerprint"].(string)),
+			Sha384Fingerprint:  utils.NewNstring(certificateDetails["sha384_fingerprint"].(string)),
+			SignatureAlgorithm: certificateDetails["signature_algorithm"].(string),
+			State:              certificateDetails["state"].(string),
+			Status:             certificateDetails["status"].(string),
+			Surname:            certificateDetails["surname"].(string),
+			Type:               certificateDetails["type"].(string),
+		}
+		CertificateDetailCollect = append(CertificateDetailCollect, servCCertificateDetails)
+		servC.CertificateDetails = CertificateDetailCollect
+	}
+	CertificateStatusList := d.Get("certificate_status").(*schema.Set).List()
+	for _, raw := range CertificateStatusList {
+		certificatestat := raw.(map[string]interface{})
+		servcCertificateStat := ov.CertificateStat{
+			ChainStatus: certificatestat["chain_status"].(string),
+			SelfSigned:  certificatestat["self_signed"].(bool),
+			Trusted:     certificatestat["trusted"].(bool),
+		}
+		servC.CertificateStatus = &servcCertificateStat
+	}
+
+	servCError := config.ovClient.CreateServerCertificate(servC)
+	var aliasname string
+	for _, servCDetail := range servC.CertificateDetails {
+		aliasname = servCDetail.AliasName
+	}
+	d.SetId(aliasname)
+	if servCError != nil {
+		d.SetId("")
+		return servCError
+	}
+	return resourceServerCertificateRead(d, meta)
+}
+
+func resourceServerCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	var servC ov.ServerCertificate
+	var err error
+	if val, ok := d.GetOk("read_type"); ok {
```

Comment:
Read type is always done by name and never by remote IP. There is no need for this read_type attribute. Read function in the resource file is called by either Create or Update and I see that in both the calls you are setting the ID with the alias_name. Read by remote IP is never used. So remove the read_type attribute, this if-else check, and call by remote IP, keep read by Name only.